File: tests/rounds-drivers/driver.pl
#!/usr/bin/perl
# Rounds-contract driver for pl/shavar.pl. See tests/rounds.sh.
use strict; use warnings; use File::Basename qw(dirname);
my $here = dirname(__FILE__);
require "$here/../../pl/shavar.pl";
my @iv = (0x6a09e667,0xbb67ae85,0x3c6ef372,0xa54ff53a,
          0x510e527f,0x9b05688c,0x1f83d9ab,0x5be0cd19);
open my $fh, '<', $ARGV[0] or die "$ARGV[0]: $!\n";
while (my $l = <$fh>) {
    next if $l =~ /^#/ || $l =~ /^\s*$/;
    my ($r) = split /\t/, $l;
    my @h = eval { main::hash_ex("abc", 24, \@iv, $r) };
    if ($@) { print "$r\trejected\t-\n" }
    else    { print "$r\taccepted\t", main::hexdigest(@h), "\n" }
}

File: tests/rounds-drivers/driver.js
// Rounds-contract driver for js/shavar.js. See tests/rounds.sh.
(function () {
  "use strict";
  var isNode = typeof process !== "undefined" && process.versions && process.versions.node;
  var argv, readFile, emit;
  if (isNode) {
    var fs = require("fs");
    argv = process.argv.slice(2);
    readFile = function (p) { return fs.readFileSync(p, "utf8"); };
    emit = function (s) { process.stdout.write(s + "\n"); };
    require(__dirname + "/../../js/shavar.js");
  } else {
    argv = arguments; readFile = read; emit = print;
    load("./js/shavar.js");
  }
  var msg = SHAVAR.hexToBytes("616263");
  readFile(argv[0]).split("\n").forEach(function (line) {
    if (!line || line.charAt(0) === "#") return;
    var r = parseInt(line.split("\t")[0], 10);
    if (isNaN(r)) return;
    try { emit(r + "\taccepted\t" + SHAVAR.bytesToHex(SHAVAR.hashEx(msg, 24, { rounds: r }))); }
    catch (e) {
      if (!(e instanceof SHAVAR.ShavarError)) throw e;
      emit(r + "\trejected\t-");
    }
  });
})();
